applications/CoSimulationApplication/custom_io/co_sim_EMPIRE_API.h

```diff
@@ -171,12 +171,26 @@ static void ReadNumberAfterKeyword(const std::string& rKeyWord, const std::strin
     line_stream >> rNumber;
 }
 
+template <typename TDataType>
+static void AllocateMemory(TDataType** ppContainer, const std::size_t Size)
+{
+    *ppContainer = new TDataType[Size];
+}
+
+template <typename TDataType>
+static void AllocateMemory(std::vector<TDataType>* pContainer, const std::size_t Size)
+{
+    if (pContainer->size() != Size) {
```

Comment:
What does it do if the current size is not equal to Size ? Should it put out a warning or something ?

Reply:
Actually the client cannot know how many nodes/elements it will receive
=> hence it would be a coincidence that the sizes are matching

Same behavior as with raw memory

Reply:
Ok ... ! 
But one more thing ... in case of using raw pointers 
there is an explicit cast to std::vector<double/int> right ?

Reply:
I don't think so...
Maybe we can take a brief look at it together

=> the idea is that you can call the fct both with std::vector (makes it easier for handling memory) or in compliance with the old Empire-API (raw-ptrs)